integrations/argocd/argocd_integration/client.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any, Optional
+import httpx
+from loguru import logger
+
+
+class ArgocdClient:
+    def __init__(self, token: str, server_url: str):
+        self.token = token
+        self.api_url = f"{server_url}/api/v1"
+        self.http_client = httpx.AsyncClient(headers=self.api_auth_header)
+
+    @property
+    def api_auth_header(self) -> dict[str, Any]:
+        return {"Authorization": f"Bearer {self.token}"}
+
+    async def _send_api_request(
+        self,
+        url: str,
+        method: str = "GET",
+        query_params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+    ) -> list[dict[str, Any]]:
+        """
+        Makes an API request to the given URL
+
+        Args:
+            url (str): API endpoint URL
+            method (str): HTTP method (default: 'GET')
+            query_params (dict): Query parameters (default: None)
+            json_data (dict): JSON data to send in request body (default: None)
+
+        Returns:
+            dict[str, Any]: A dictionary containing the JSON response from the resource.
+        """
+        try:
+            response = await self.http_client.request(
+                method=method, url=url, params=query_params, json=json_data
+            )
+
+            response.raise_for_status()
+            return response.json()["items"]
```

Comment:
I think the logic of parsing the response json shouldn't be in this method but rather in the method calling it/or in the parser, we want this to be agnostic as possible so if we will need to add another api to query we won't need to change this method because the response is different
```suggestion
            return response.json()
```